core/server/worker/src/main/java/alluxio/worker/block/meta/StorageTier.java

```diff
@@ -235,7 +235,7 @@ public StorageDir getDir(int dirIndex) {
    * @return a list of lost storage paths
    */
   public List<String> getLostStorage() {
-    return Collections.unmodifiableList(mLostStorage);
+    return new ArrayList<>(mLostStorage);
```

Comment:
can we update `getStorageDirs()` to also return a copy? It has the same issue

Reply:
Now the other places except the `BlockHeartbeatReport` all returns a copy of `lost storages`